components/concept/sync/src/main/java/mozilla/components/concept/sync/OAuthAccount.kt

```diff
@@ -33,20 +33,155 @@ data class AuthFlowUrl(val state: String, val url: String)
+
+    /**
+     * Returns current FxA Device ID for an authenticated account.
+     *
+     * @return Current device's FxA ID, if available. `null` otherwise.
+     */
     fun getCurrentDeviceId(): String?
+
+    /**
+     * Returns session token for an authenticated account.
+     *
+     * @return Current account's session token, if available. `null` otherwise.
+     */
     fun getSessionToken(): String?
+
+    /**
+     * Returns an authorized auth code for the passed scopes
+     *
+     * @param clientId the client id string
+     * @param scopes the list of scopes to request access to
+     * @param state the state token string
+     * @param accessType the accessType method to be used by the returned code, "offline" or "online"
+     * @return the authorized auth code string
+     */
+    fun authorizeOAuthCode(
+        clientId: String,
+        scopes: Array<String>,
+        state: String,
+        accessType: String = "online"
+    ): String?
+
+    /**
+     * Fetches the profile object for the current client either from the existing cached account,
+     * or from the server (requires the client to have access to the profile scope).
+     *
+     * @param ignoreCache Fetch the profile information directly from the server
+     * @return Profile (optional, if successfully retrieved) representing the user's basic profile info
+     */
     fun getProfileAsync(ignoreCache: Boolean): Deferred<Profile?>
+
+    /**
+     * Convenience method to fetch the profile from a cached account by default, but fall back
+     * to retrieval from the server.
+     *
+     * @return Profile (optional, if successfully retrieved) representing the user's basic profile info
+     */
     fun getProfileAsync(): Deferred<Profile?>
+
+    /**
+     * Authenticates the current account using the code and state parameters fetched from the
+     * redirect URL reached after completing the sign in flow triggered by [beginOAuthFlowAsync].
+     *
+     * Modifies the FirefoxAccount state.
+     * @param code OAuth code string
+     * @param state state token string
+     * @return Deferred boolean representing success or failure
+     */
     fun completeOAuthFlowAsync(code: String, state: String): Deferred<Boolean>
+
+    /**
+     * Tries to fetch an access token for the given scope.
+     *
+     * @param singleScope Single OAuth scope (no spaces) for which the client wants access
+     * @return [AccessTokenInfo] that stores the token, along with its scope, key and
+     *                           expiration timestamp (in seconds) since epoch when complete
+     */
     fun getAccessTokenAsync(singleScope: String): Deferred<AccessTokenInfo?>
+
+    /**
+     * This method should be called when a request made with an OAuth token failed with an
+     * authentication error. It will re-build cached state and perform a connectivity check.
+     *
+     * This will use the network.
```

Comment:
Remove this comment, or add it to every other kdoc. Most of these methods will use the network.